explorer/ast/member.cpp

```diff
@@ -14,19 +14,34 @@ Member::Member(Nonnull<const Declaration*> declaration)
 Member::Member(Nonnull<const NamedValue*> struct_member)
     : member_(struct_member) {}
 
+Member::Member(Nonnull<const IndexedValue*> tuple_member)
+    : member_(tuple_member) {}
+
 auto Member::name() const -> std::string_view {
   if (const auto* decl = member_.dyn_cast<const Declaration*>()) {
     return GetName(*decl).value();
+  } else if (const auto* named_valued = member_.dyn_cast<const NamedValue*>()) {
+    return named_valued->name;
+  } else {
+    return "";
```

Comment:
I think returning an empty string here is likely to cause bugs; at a minimum I'd like this to `CHECK` fail instead, but ideally I'd like to restructure the API so you can't write those bugs in the first place. Specifically, I think we should get rid of `name()` on both `Member` and `Component`, and replace it with two separate operations:

- `void Print(llvm::raw_ostream& out)`, which prints the `Member`/`Component`.
- `auto IsNamed(std::string_view name) -> bool`, which returns true if the `Member`/`Component` is named `name`.

I think those two are enough to support all the current use cases for `name`.

Reply:
Thank you, I did add both methods, but had to keep `name()` for now, because `GetMember` (now `GetNamedMember`) relies heavily on having a string version of the name available for various types and situation, provided to multiple methods. So that wasn't actually practical due to that one method content.